In [1]:
import MeCab
import unidic
print(unidic.DICDIR)
MeCab.Tagger("-d" + unidic.DICDIR)

/home/yamada/.local/lib/python3.6/site-packages/unidic/dicdir


<Swig Object of type 'MeCab::Tagger *' at 0x7f690827c6d0>

- よくわからないもの

    
    

In [2]:
import json
import os
import pandas as pd
import numpy as np
from pathlib import Path
import collections
from sklearn.model_selection import train_test_split
from sklearn import metrics

import sys
sys.path.append("../")
from datatools.analyzer import *
from utterance.error_tools import *

from datatools.maneger import DataManager
from datatools.preproc import Preprocessor

In [3]:
ngram_path = "../../corpus/google/3-gram/"

In [4]:
key1_set = set( os.listdir(ngram_path) ) 

In [17]:
path = "../eval_labeled/"
datalist = ['DCM', 'DIT', 'IRS']
convs = read_conv(path, datalist)

error = "Grammatical error"
# errors = ['Grammatical error', "Uninterpretable"]
errors = ['Semantic error', "Wrong information"]
sys_utt = []
y = []
for conv in convs:
    for ut in conv:
        if ut.is_system() and ut.is_exist_error():
            if not ut.utt[-1] in ["？", "！", "。", "!"]:
                sys_utt.append( clean_text( ut.utt+"。" ))
                # sys_utt.append(ut.utt)
            else:   
                sys_utt.append(ut.utt)
            if ut.is_error_included(errors):
                y.append(1)
            else:
                y.append(0)

In [6]:
print("len of y:{0}, error '{1}' counts:{2}".format(len(y), errors, y.count(1)))

len of y:1386, error '['Grammatical error', 'Uninterpretable']' counts:7


In [7]:
def load_dict(path, key1):
    filename = key_formated(key1)
    with open(path+filename, "r") as f:
        key1_dict = json.load(f) 
    return key1_dict  

In [8]:
def key_formated(key1):
    return "ngram_"+key1+".json"

In [9]:
key2dict = dict()
registerform_dict = dict()
analyzer_unidic=  Analyzer(Tokenizer())

In [10]:
#  を考慮
# 何故か↑が出てくる
def judge_key3(key1_dict, ngram):
    key1 = ngram[0]
    key2 = ngram[1]
    key3 = ngram[2]

    if key2 not in key1_dict[key1]:
        # '"'をkeyにして聞いてみる
        if '"' in key1_dict[key1] and key2 in key1_dict[key1]['"']:
            return True
        if '、' in key1_dict[key1] and key2 in key1_dict[key1]['、']:
            return True
        return False
    
    if key3 not in key1_dict[key1][key2]:
        return False

    return True 

In [11]:

def search_registerforam(key, reading):
    filename = key_formated(key)
    if filename in key1_set:
        return key
    # 登録されていない
    else:
        for other in reading:
            filename = key_formated(key)
            if filename in key1_set:
                registerform_dict[key] = other
                return other
        return key


def sentence_judge(sentence, N=3):
    
    df = analyzer_unidic.analyze_with_dataframe(sentence)
    
    # 登録
    sentence_ = ""
    for txt, read in zip(df.surface, df.reading):
        sentence_ += search_registerforam(txt, read)
    sents = sentence2docs(sentence_, sents_span=True)
    # print(sents)

    # print(tokens)
    ngram_text = []
    ngram_pos = []

    for sent in sents:
        # 各文章 : sent
        df = analyzer_unidic.analyze_with_dataframe("".join([token.orth_ for token in sent]))
        surfaces = []
        poses = []
        readings = []
        for txt, p, read in zip(df.surface, df.part_of_speech, df.reading):
            surfaces.append(txt)
            readings.append(read)
            poses.append(p)

            for i in range(len(surfaces)-N+1):
                # print(L[i:i+N])
                ngram_text.append(surfaces[i:i+N])
                ngram_pos.append(poses[i:i+N])
                # ngram_reading.append(readings[i:i+N])
    
    # print(ngram_text)
    # return
    
    for ngram, pgram in zip(ngram_text, ngram_pos):
        # if "助動詞" in pgram[1] or "助詞" in pgram[1] or "助動詞" in pgram[0] or "助詞" in pgram[0]:
        # if "助動詞" in pgram[0] or "助詞" in pgram[0]:
        if "助詞" in pgram[1]:
            # print("\tcontext : {0}| ->".format(ngram))

            key1 = ngram[0]
            if key_formated(key1) not in key1_set:
                continue

            if key1 in key2dict:
                key1_dict = key2dict[key1]
            else:
                key1_dict = load_dict(ngram_path, key1)
                key2dict[key1] = key1_dict

            # チェック
            if not judge_key3(key1_dict, ngram):
                print(ngram)
                # reading を調整
                
                # 見つかれば，その語彙を登録
                return False
    return True
            


In [12]:
y_pred_2 = []
for i, utt in enumerate( tqdm( sys_utt ) )  :
    cleaned = clean_text(utt).replace("?", "。").replace("!", "")
    # print(cleaned)
    if sentence_judge(cleaned, N=3):
        y_pred_2.append(0)
    else:
        y_pred_2.append(1)
    # if i==35:
    #     break

  0%|          | 2/1386 [00:00<08:18,  2.78it/s]

['遍路', 'の', '逆回り']


  4%|▍         | 55/1386 [00:20<03:04,  7.20it/s] 

['天麩羅', 'は', '食べる']
['寿司', 'は', '縁側']
['なん', 'と', '。']


  5%|▌         | 70/1386 [00:25<03:51,  5.68it/s]

['渾名', 'は', 'あり']
['渾名', 'は', 'あり']
['渾名', 'を', '持ち']
['同僚', 'を', '渾名']
['同僚', 'を', '渾名']


  5%|▌         | 74/1386 [00:25<02:36,  8.36it/s]

['掛け声', 'が', '出す']
['掛け声', 'を', '覚える']
['大声', 'で', '上げる']


  6%|▌         | 85/1386 [00:26<01:25, 15.21it/s]

['映画', 'は', '一人']


  6%|▋         | 88/1386 [00:26<01:25, 15.24it/s]

['位', 'ヶ', '岳']
['位', 'ヶ', '岳']
['デート', 'で', '御洒落']


  7%|▋         | 94/1386 [00:26<01:30, 14.26it/s]

['勉強', 'も', '頑張ろう']
['フラワー', 'は', '手芸']


  7%|▋         | 97/1386 [00:26<01:18, 16.46it/s]

['感じる', 'ん', 'でしょう']


  8%|▊         | 104/1386 [00:27<01:09, 18.36it/s]

['外食', 'は', 'い']
['独身', 'で', '0']
['由紀恵', 'は', '怖い']


  8%|▊         | 110/1386 [00:31<08:17,  2.57it/s]

['デート', 'で', '御洒落']


  8%|▊         | 112/1386 [00:31<07:06,  2.99it/s]

['争い', 'が', '起ころう']


  8%|▊         | 114/1386 [00:33<11:26,  1.85it/s]

['と', 'か', '泣き']


  8%|▊         | 117/1386 [00:34<08:01,  2.64it/s]

['なら', 'を', '使う']
['池', 'は', '寒い']


  9%|▉         | 126/1386 [00:34<02:49,  7.45it/s]

['たい', 'ん', 'でしょう']


 10%|▉         | 135/1386 [00:35<01:57, 10.63it/s]

['てる', 'ん', 'でしょう']
['微熱', 'は', '多い']


 11%|█         | 149/1386 [00:37<02:06,  9.80it/s]

['汗', 'は', 'かっこ']


 11%|█▏        | 159/1386 [00:39<02:38,  7.74it/s]

['さん', 'の', '朝御飯']
['宇宙', 'で', 'プレー']


 12%|█▏        | 161/1386 [00:39<02:16,  8.95it/s]

['パー', 'より', '0']


 12%|█▏        | 168/1386 [00:39<01:45, 11.50it/s]

['時期', 'から', '資格']


 13%|█▎        | 176/1386 [00:40<01:14, 16.19it/s]

['最近', 'じゃあ', '漫画']
['干支', 'は', '辰年']
['です', 'か', '…']


 13%|█▎        | 185/1386 [00:40<01:08, 17.63it/s]

['本当', 'に', 'ラーメン']
['補給', 'は', '少ない']


 14%|█▍        | 193/1386 [00:41<00:54, 21.72it/s]

['料理', 'は', '手早い']


 14%|█▍        | 200/1386 [00:41<00:54, 21.91it/s]

['つける', 'と', 'ホッと']


 15%|█▌        | 208/1386 [00:41<00:43, 27.33it/s]

['料理', 'は', '手早い']
['独り暮らし', 'は', '食']


 16%|█▌        | 218/1386 [00:42<00:48, 24.03it/s]

['手', 'を', '貯金']
['手', 'を', '貯金']


 16%|█▌        | 225/1386 [00:42<00:55, 20.81it/s]

['頬', 'に', 'ピアス']
['耳朶', 'の', '大きい']
['金', 'は', '大き']
['お', 'や', '、']


 17%|█▋        | 234/1386 [00:44<02:33,  7.49it/s]

['スイカ', 'は', '面白い']


 19%|█▊        | 257/1386 [00:45<00:48, 23.44it/s]

['料理', 'は', 'そろい']
['一人', 'で', '食べる']


 20%|██        | 279/1386 [00:46<00:32, 34.18it/s]

['ウェブ', 'と', 'ホーム']
['梅雨入り', 'は', 'いい']
['カタ', 'を', '使い']


 21%|██        | 293/1386 [00:46<00:28, 38.05it/s]

['料理', 'は', '手早い']
['泳ぐ', 'を', '覚える']
['泳ぐ', 'を', '繰り返す']
['泳ぐ', 'を', '覚え']
['泳ぐ', 'を', '覚え']
['泳ぐ', 'を', '覚える']


 22%|██▏       | 302/1386 [00:47<00:28, 38.21it/s]

['バニラ', 'は', '相性']
['た', 'を', '仕入れる']


 23%|██▎       | 312/1386 [00:47<00:29, 36.86it/s]

['た', 'は', '鮮度']


 23%|██▎       | 320/1386 [00:47<00:32, 32.47it/s]

['嗽', 'で', '予防']
['嗽', 'で', '予防']
['心掛ける', 'か', 'も']


 24%|██▎       | 329/1386 [00:48<00:42, 24.67it/s]

['天辺', 'を', '軽く']
['天辺', 'を', '軽く']
['ID', 'に', '行き']
['ID', 'に', '行き']


 24%|██▍       | 337/1386 [00:48<00:38, 27.33it/s]

['求める', 'か', 'も']
['せき', 'が', '欲しい']
['恋人', 'は', '重い']


 25%|██▍       | 341/1386 [00:48<00:46, 22.58it/s]

['本当', 'に', 'おもしろい']


 26%|██▌       | 356/1386 [00:49<00:50, 20.32it/s]

['天気', 'は', '恋しい']


 26%|██▌       | 362/1386 [00:49<00:52, 19.46it/s]

['教える', 'か', 'も']


 27%|██▋       | 371/1386 [00:50<00:58, 17.27it/s]

['摘み', 'は', 'おいしい']


 27%|██▋       | 378/1386 [00:50<00:52, 19.11it/s]

['感', 'って', 'すっきり']
['感', 'って', 'すっきり']
['デート', 'で', '御洒落']


 28%|██▊       | 385/1386 [00:51<01:00, 16.47it/s]

['アクセス', 'の', 'ステー']
['アクセス', 'が', '都心']
['日本', 'と', 'JR']


 28%|██▊       | 391/1386 [00:51<00:53, 18.55it/s]

['楽しむ', 'と', '花']


 29%|██▉       | 400/1386 [00:52<00:56, 17.34it/s]

['ハーバー', 'の', '0']
['なっ', 'しー', 'が']


 29%|██▉       | 402/1386 [00:52<01:33, 10.53it/s]

['商品', 'に', '今日']
['テレビ', 'は', 'サイド']


 29%|██▉       | 407/1386 [00:53<01:43,  9.44it/s]

['ジブリ', 'の', '踊る']
['ジブリ', 'の', '隣り']
['のぶ代', 'と', 'ハビエル']
['男', 'を', '立体']


 30%|██▉       | 409/1386 [00:53<01:51,  8.79it/s]

['つとめ', 'て', '今年']
['今', 'の', 'フォーブス']


 30%|██▉       | 411/1386 [00:53<01:54,  8.51it/s]

['熱', 'の', '緊急']
['アメリカ', 'で', 'ジャイアン']


 30%|███       | 416/1386 [00:54<01:55,  8.39it/s]

['先行き', 'に', '対する']
['IQ', 'と', 'いえ']
['映画', 'の', 'You']
['鎌倉', 'の', '全国']
['鎌倉', 'の', '笑点']
['鎌倉', 'の', 'ファミリー']


 30%|███       | 419/1386 [00:54<01:27, 11.04it/s]

['マート', 'が', '配送']
['家', 'に', '中井']
['鎌倉', 'の', '馬鹿']


 30%|███       | 422/1386 [00:55<01:21, 11.84it/s]

['サイト', 'の', 'ダビンチ']
['サイト', 'の', 'あま']


 31%|███       | 433/1386 [00:57<02:10,  7.29it/s]

['饂飩', 'の', '御']
['饂飩', 'の', '御']
['金団', 'で', 'ソフト']
['バンク', 'の', '大人']


 32%|███▏      | 438/1386 [00:57<01:26, 11.01it/s]

['ロボット', 'の', 's']
['バンク', 'の', '家庭']
['バンク', 'の', '家庭']
['バンク', 'の', '家庭']
['空輸', 'の', '話題']


 32%|███▏      | 442/1386 [00:58<01:28, 10.70it/s]

['山', 'の', '海遊']
['海月', 'が', '大好き']
['館内', 'の', '太陽']


 32%|███▏      | 447/1386 [00:58<01:10, 13.39it/s]

['ホワイト', 'と', 'カフェ']
['パッケージ', 'も', '御洒落']
['大阪', 'の', 'チョコレートラテポッキーブレンド']
['ゲゲゲ', 'の', '女房']
['ゲゲゲ', 'の', '女房']


 32%|███▏      | 449/1386 [00:58<01:08, 13.78it/s]

['保険', 'を', '観戦']
['連盟', 'の', 'ギャグ']
['菓子', 'の', '鶴']


 33%|███▎      | 453/1386 [00:58<01:12, 12.96it/s]

['旅', 'の', '指差し']
['味', 'の', 'shall']


 33%|███▎      | 457/1386 [00:59<01:04, 14.32it/s]

['蜜柑', 'に', '比べ']
['来年', 'に', 'プロ']
['用', 'に', '饂飩']
['ソックス', 'の', 'キャンデー']


 33%|███▎      | 459/1386 [00:59<01:04, 14.36it/s]

['ソックス', 'の', 'キャンデー']
['野球', 'の', 'Microsoft']
['Microsoft', 'の', '里田']


 33%|███▎      | 461/1386 [00:59<01:40,  9.21it/s]

['投手', 'が', '切っ掛け']
['ウェンリー', 'と', 'いう']
['だ', 'から', '荒野']


 33%|███▎      | 464/1386 [00:59<01:26, 10.61it/s]

['料理', 'を', '受賞']
['街', 'に', 'NHK']


 34%|███▍      | 469/1386 [01:00<01:14, 12.23it/s]

['NHK', 'の', '町歩き']
['レボリューション', 'は', 'NHK']


 34%|███▍      | 473/1386 [01:00<01:09, 13.08it/s]

['協会', 'など', '0']
['デュボア', 'の', '授賞']
['グランプリ', 'の', 'レプリカ']
['ゲゲゲ', 'の', '女房']
['ゲゲゲ', 'の', '女房']


 34%|███▍      | 476/1386 [01:05<09:06,  1.67it/s]

['は', 'は', 'ハードル']


 35%|███▍      | 481/1386 [01:05<04:49,  3.12it/s]

['NHK', 'の', '町歩き']
['レント', 'の', '永作']
['保険', 'が', '連載']


 35%|███▍      | 485/1386 [01:06<03:35,  4.19it/s]

['イングレス', 'の', 'レプリカ']
['イングレス', 'と', 'コンビニエンス']
['マクドナルド', 'の', '遠藤']


 35%|███▌      | 490/1386 [01:06<02:13,  6.73it/s]

['問題', 'で', 'マクドナルド']
['マクドナルド', 'で', 'リング']
['マクドナルド', 'の', 'リング']
['公園', 'と', '遠刈田']
['NHK', 'の', 'ローリー']


 36%|███▌      | 496/1386 [01:07<01:44,  8.50it/s]

['ゲゲゲ', 'の', '女房']
['NHK', 'の', '町歩き']
['実際', 'は', 'タモリ']
['ランド', 'や', '東京']


 36%|███▌      | 502/1386 [01:07<01:08, 12.91it/s]

['バレーボール', 'の', '八木']
['ディズニーシー', 'の', '職人']
['なっ', 'しー', 'の']
['チュウ', 'も', '使用']
['最近', 'は', 'ヨーク']


 36%|███▋      | 505/1386 [01:07<01:00, 14.57it/s]

['ヒルズ', 'と', 'JR']
['ヒルズ', 'から', 'さほど']
['ヒルズ', 'で', '000000000']
['アナウンサー', 'を', '御']
['ダッシュ', 'など', '多く']


 37%|███▋      | 508/1386 [01:07<00:53, 16.51it/s]

['テレビ', 'の', '林家']


 37%|███▋      | 512/1386 [01:08<01:40,  8.67it/s]

['流れ', 'を', '関西']
['NHK', 'の', 'ケーキ']


 37%|███▋      | 519/1386 [01:09<01:48,  8.00it/s]

['女子', 'の', 'ブーム']
['だ', 'から', '荒野']
['あっ', 'で', 'セアカゴケグモ']
['レント', 'の', 'タモリ']


 38%|███▊      | 525/1386 [01:09<01:04, 13.30it/s]

['小説', 'の', 'ブラタモリ']
['弦', 'は', '奈良']
['奈良', 'の', '閻']
['元気', 'や', 'を']
['タイガー', 'の', 'や']
['タイガー', 'の', 'デギュスタシオン']


 38%|███▊      | 528/1386 [01:09<00:58, 14.78it/s]

['スケート', 'の', '東京']
['恵比寿', 'の', '画家']
['年始', 'に', '泊まり']
['野薔薇', 'か', 'も']
['野薔薇', 'か', 'も']


 39%|███▊      | 534/1386 [01:10<00:51, 16.44it/s]

['野薔薇', 'か', 'もん']
['シリーズ', 'に', 'カレー']
['味', 'も', 'シーフード']


 39%|███▊      | 537/1386 [01:10<00:45, 18.59it/s]

['味', 'の', 'アムゼル']
['で', 'は', '日清']
['シリーズ', 'の', '濃厚']


 39%|███▉      | 543/1386 [01:10<00:57, 14.74it/s]

['戦', 'と', 'ホンジュラス']
['プレー', 'が', 'シドニー']
['パリーグ', 'に', '入団']
['球団', 'の', 'パリーグ']
['投手', 'の', 'シドニー']
['球団', 'の', 'パシフィック']


 40%|███▉      | 549/1386 [01:11<01:01, 13.53it/s]

['シドニー', 'の', 'ベンチャー']
['シドニー', 'の', '慣習']
['コロッケ', 'が', '自動']
['陽平', 'や', 'アルベルト']
['オランダ', 'で', '展示']


 40%|███▉      | 554/1386 [01:11<00:51, 16.17it/s]

['代表', 'の', 'アムステルダム']
['ポスト', 'の', 'joy']
['車', 'と', '有名']
['0', 'の', '木村']


 40%|████      | 558/1386 [01:12<01:06, 12.38it/s]

['クッキー', 'を', '踊り']
['明子', 'と', 'ローリー']


 40%|████      | 560/1386 [01:12<01:01, 13.43it/s]

['薯', 'の', '御薩']
['ポテト', 'の', 'ちょい']


 41%|████      | 566/1386 [01:12<01:04, 12.77it/s]

['スイーツ', 'は', 'どの']
['0', 'の', 'ドラマ']


 41%|████      | 568/1386 [01:12<01:06, 12.22it/s]

['し', 'て', '000000000']
['階', 'の', '劇団']


 41%|████▏     | 573/1386 [01:13<01:09, 11.73it/s]

['ミュージカル', 'は', '特別']
['劇場', 'に', 'インスパイアー']
['四季', 'の', '岩田']
['四季', 'の', '朝日']
['劇場', 'と', '旭山']


 42%|████▏     | 579/1386 [01:13<00:48, 16.75it/s]

['園', 'の', 'オーレ']
['海豹', 'が', '好き']
['カルビー', 'の', '海豹']
['漁師', 'の', '力め']
['園', 'の', 'チョコレート']
['園', 'も', '大体']


 42%|████▏     | 582/1386 [01:13<00:47, 16.76it/s]

['県', 'が', '桜ん坊']
['スシロー', 'は', '山形']
['ソックス', 'は', '鶴岡']


 42%|████▏     | 586/1386 [01:14<00:50, 15.86it/s]

['内', 'で', 'ツキノワ']
['暮らし', 'の', 'e']
['チュウ', 'で', '送料']
['グランプリ', 'で', '送料']


 43%|████▎     | 592/1386 [01:14<00:45, 17.38it/s]

['たい', 'の', 'でしょう']
['公園', 'と', 'ナゴヤ']
['グリコ', 'の', '名古屋']
['グリコ', 'の', '安納']
['グリコ', 'の', '安納']


 43%|████▎     | 598/1386 [01:14<00:40, 19.25it/s]

['ストップ', 'の', 'ファミマ']
['ストップ', 'の', '筋肉']
['ストップ', 'と', 'パステル']
['クリーム', 'で', '思い出し']
['流', 'の', '羽生']


 43%|████▎     | 601/1386 [01:14<00:41, 18.91it/s]

['戦', 'と', 'ホンジュラス']
['MVP', 'は', '前']
['さん', 'が', '二人']


 44%|████▎     | 606/1386 [01:15<00:59, 13.21it/s]

['投手', 'へ', 'メジャー']
['投手', 'は', 'キリン']
['キリン', 'の', 'サンタ']
['クロース', 'の', '御']


 44%|████▍     | 609/1386 [01:15<00:58, 13.38it/s]

['映画', 'の', 'ミスター']
['パリーグ', 'の', 'ミスター']
['球場', 'まで', '駆け付け']


 44%|████▍     | 613/1386 [01:16<01:02, 12.41it/s]

['東京', 'で', '鶴']
['ドール', 'に', '代表']
['大阪', 'の', '馬事']


 45%|████▍     | 619/1386 [01:16<00:44, 17.25it/s]

['マニア', 'は', '東京']
['春休み', 'で', 'とても']
['連合', 'の', 'キャラクター']
['連合', 'の', 'ギャグ']


 45%|████▍     | 622/1386 [01:16<01:03, 11.94it/s]

['シーク', 'に', 'ノミネート']
['歌手', 'の', '明石家']
['芸人', 'の', '大竹']
['あなた', 'が', 'リオネル']
['マリノス', 'の', 'リオネル']


 45%|████▌     | 625/1386 [01:16<00:55, 13.67it/s]

['泰幸', 'も', '現在']
['マリノス', 'の', '今野']


 45%|████▌     | 630/1386 [01:17<00:51, 14.67it/s]

['マリノス', 'で', 'キャプテンシー']
['ヤマザキ', 'は', '天皇']
['静岡', 'の', 'じゃがりこ']
['lush', 'は', '、']
['lush', 'と', 'は']
['宇治', 'で', '販売']


 46%|████▌     | 636/1386 [01:17<00:42, 17.58it/s]

['カルビー', 'の', '約']
['立ち寄っ', 'て', 'ハロー']
['系列', 'の', 'ずっと']
['系列', 'の', 'ずっと']


 46%|████▌     | 640/1386 [01:17<00:47, 15.63it/s]

['プロポーズ', 'の', '年越し']
['系列', 'の', 'ザ']
['妻', 'を', '一筋']
['番組', 'の', '御飯']


 46%|████▋     | 644/1386 [01:18<00:47, 15.71it/s]

['喜劇', 'で', '放送']
['喜劇', 'の', '明石家']
['本当', 'は', 'こんな']
['さんま', 'の', '侭']


 47%|████▋     | 647/1386 [01:18<00:41, 17.82it/s]

['静岡', 'と', 'マドリード']
['キー', 'の', '抹茶']
['味', 'は', 'You']
['静岡', 'の', '水曜']


 47%|████▋     | 649/1386 [01:18<00:42, 17.53it/s]

['土産', 'の', 'あまおう']
['社', 'か', 'ループ']


 47%|████▋     | 651/1386 [01:18<00:51, 14.31it/s]

['。', 'の', '収録']
['楽曲', 'を', '著名']


 47%|████▋     | 655/1386 [01:18<00:55, 13.21it/s]

['など', 'が', 'You']
['ベースボール', 'に', '入団']
['投手', 'が', 'シンガポール']


 48%|████▊     | 660/1386 [01:19<00:43, 16.63it/s]

['シンガポール', 'の', '東北']
['東北', 'の', '松坂']


 48%|████▊     | 664/1386 [01:19<00:43, 16.79it/s]

['ライオンズ', 'の', 'カミング']
['沖縄', 'の', 'カミング']
['赤', 'の', '漆塗り']


 48%|████▊     | 668/1386 [01:19<00:45, 15.73it/s]

['安室', 'の', 'オリオン']
['で', 'は', '鸚哥']


 49%|████▊     | 674/1386 [01:20<00:48, 14.54it/s]

['級', 'の', '水族']
['リーズ', 'の', 'イエス']
['アギーレ', 'の', '誕生']
['代表', 'の', 'EU']


 49%|████▉     | 681/1386 [01:20<00:47, 14.87it/s]

['タイム', 'や', '冬']
['パスポート', 'の', '海豹']


 49%|████▉     | 685/1386 [01:20<00:37, 18.93it/s]

['園', 'の', 'バター']
['海豹', 'を', '間近']
['ドラマ', 'の', '黄色い']
['コロッケ', 'より', 'オランダ']


 50%|████▉     | 688/1386 [01:20<00:35, 19.94it/s]

['ゴッホ', 'や', 'ハビエル・アギーレ']
['アギーレ', 'や', 'フェルメール']
['味', 'の', 'ココ']


 50%|█████     | 694/1386 [01:21<00:44, 15.48it/s]

['情報', 'を', '御洒落']
['オランダ', 'で', '展示']
['テレビ', 'の', 'メロン']


 50%|█████     | 698/1386 [01:21<00:54, 12.73it/s]

['タイガース', 'の', 'BANDAI']
['ファイターズ', 'の', 'BANDAI']
['中止', 'で', '思い出し']
['パン', 'は', 'スーパーマーケット']


 51%|█████     | 702/1386 [01:22<00:50, 13.54it/s]

['皮', 'より', '重量']


 51%|█████     | 706/1386 [01:23<01:52,  6.03it/s]

['受け取り', 'が', '元気']
['ジャパン', 'の', 'ハンバーガー']
['アナ', 'と', '雪']
['州', 'と', 'スペイン']


 51%|█████     | 710/1386 [01:23<01:14,  9.10it/s]

['スペイン', 'や', 'マドリード']
['ストア', 'の', '湘南']
['ストア', 'の', '湘南']
['菓子', 'の', 'リプトン']


 52%|█████▏    | 715/1386 [01:23<00:48, 13.91it/s]

['0', 'の', 'スキー']
['選手', 'の', 'ドゥンガ']
['チュウ', 'の', 'レプリカ']
['未だ', 'に', 'アップル']


 52%|█████▏    | 721/1386 [01:23<00:36, 18.04it/s]

['八幡平', 'より', 'もう']
['土産', 'に', '島歌']
['ガーデン', 'の', '島歌']
['今年', 'の', 'グラッチェ']


 52%|█████▏    | 724/1386 [01:24<00:37, 17.74it/s]

['種類', 'だけ', 'メニュー']
['し', 'ながら', 'みけ']
['台湾', 'と', '東北']
['ソックス', 'の', '事故']


 53%|█████▎    | 730/1386 [01:24<00:38, 16.87it/s]

['時代', 'に', '御嶽']
['とんねるず', 'の', '皆']
['ノンストップ', 'も', '、']
['テレビ', 'の', '博士']


 53%|█████▎    | 739/1386 [01:24<00:30, 21.40it/s]

['菓子', 'の', 'リプトン']
['奈生子', 'の', '絵画']
['大阪', 'の', 'ミッドフィルダー']
['ストア', 'の', 'セレッソ']
['大阪', 'は', '今期']
['札幌', 'の', '長友']


 54%|█████▍    | 746/1386 [01:25<00:37, 16.90it/s]

['誠', 'が', 'コンサドーレ']
['。', 'の', '他']
['マート', 'の', 'リーダー']
['半島', 'の', 'You']
['←', 'の', 'You']


 54%|█████▍    | 749/1386 [01:25<00:38, 16.35it/s]

['キャンデー', 'で', '見']
['俳優', 'の', 'ヒカ']
['アーチスト', 'は', 'どんな']
['の', 'が', 'ツイッター']
['ツイッター', 'が', '漫画']


 54%|█████▍    | 755/1386 [01:25<00:32, 19.23it/s]

['フール', 'に', 'ツイッター']
['茅ヶ崎', 'と', 'アムステルダム']
['スターズ', 'の', '曲']
['バンド', 'の', 'サザン']
['FC', 'の', '楽曲']


 55%|█████▍    | 758/1386 [01:26<00:30, 20.66it/s]

['活躍', 'で', '茅ヶ崎']
['FC', 'の', '楽曲']
['ひろ子', 'と', '鉄拳']


 55%|█████▍    | 761/1386 [01:26<00:35, 17.60it/s]

['私', 'も', 'ブラタモリ']
['NHK', 'の', '町歩き']


 55%|█████▌    | 767/1386 [01:26<00:37, 16.72it/s]

['ピッチング', 'の', '安心']
['横綱', 'は', '海外']
['さび', 'で', '考え']


 56%|█████▌    | 773/1386 [01:27<00:35, 17.20it/s]

['映画', 'の', '宇多田']
['英', 'の', 'オーディション']
['タレント', 'の', '収録']
['台湾', 'の', '宇多田']


 56%|█████▌    | 776/1386 [01:27<00:32, 18.92it/s]

['サカナクション', 'の', '他']
['トキオ', 'の', 'ライオン']
['ボランチ', 'は', '栗原']


 56%|█████▋    | 782/1386 [01:27<00:32, 18.87it/s]

['料理', 'の', '牛蒡']
['大阪', 'の', 'あさイチ']
['サッカー', 'の', '川澄']
['選手', 'の', '御陰']


 57%|█████▋    | 785/1386 [01:27<00:33, 18.12it/s]

['大阪', 'の', 'アンデルソン']
['山形', 'も', '勢い']


 57%|█████▋    | 787/1386 [01:27<00:34, 17.28it/s]

['省', 'が', '総額']


 57%|█████▋    | 793/1386 [01:28<00:42, 14.11it/s]

['院', 'の', 'ソレデモシタイ']
['院', 'の', 'バラード']
['稼働', 'なんて', '、']


 57%|█████▋    | 795/1386 [01:28<00:41, 14.33it/s]

['朝日', 'の', '石井']
['朝日', 'の', '出入']
['系列', 'の', 'さんま']


 58%|█████▊    | 799/1386 [01:28<00:42, 13.84it/s]

['テニス', 'の', '矢部']
['浩之', 'は', '人気']
['ナイン', 'で', '放送']
['番組', 'の', 'ぶー']
['さん', 'の', 'ぶー']


 58%|█████▊    | 804/1386 [01:29<00:36, 15.90it/s]

['館', 'と', '国営']
['県', 'が', '桜ん坊']
['スーパーマーケット', 'の', 'や']


 58%|█████▊    | 807/1386 [01:29<00:31, 18.12it/s]

['フーズ', 'の', 'グループ']
['年', 'の', 'ソントン']


 59%|█████▊    | 811/1386 [01:29<00:34, 16.53it/s]

['楽曲', 'の', 'タイムズ']
['スマップ', 'の', '浜崎']
['さん', 'か', '宇多田']
['beta', 'に', '参加']


 59%|█████▉    | 820/1386 [01:30<00:53, 10.65it/s]

['楽曲', 'を', '著名']
['Tube', 'の', 'プロモーション']
['芸人', 'の', 'マツコ']
['芸人', 'の', 'マツコ']


 59%|█████▉    | 823/1386 [01:30<00:44, 12.60it/s]

['商品', 'を', 'マツコ']
['俳優', 'の', 'マツコ']
['マツコ', 'の', '知ら']
['俳優', 'の', 'マツコ']
['グリコ', 'の', '鸚哥']


 60%|█████▉    | 828/1386 [01:30<00:35, 15.51it/s]

['グリコ', 'の', 'グランド']
['グリコ', 'の', 'ぐり']
['グリコ', 'の', '竹の子']
['グリコ', 'の', '苺']


 60%|██████    | 834/1386 [01:31<00:28, 19.68it/s]

['ラインアップ', 'の', '中']
['ポッキー', 'が', '大阪']
['グリコ', 'の', '広島']
['園', 'と', '横川']


 61%|██████    | 839/1386 [01:31<00:30, 18.23it/s]

['ビール', 'の', 'ハッピー']
['ビール', 'の', 'スナック']
['自宅', 'で', 'アサヒ']
['ビール', 'の', 'スナック']
['ビール', 'の', 'かっぱ']


 61%|██████    | 842/1386 [01:31<00:26, 20.25it/s]

['選手', 'の', '三都主']
['レプリカ', 'が', '懸賞']
['コロッケ', 'は', '自動']
['日本', 'の', 'モンテディオ']


 61%|██████    | 848/1386 [01:31<00:28, 18.83it/s]

['コメ', 'を', 'ブラジル']
['。', 'が', '世界']
['タイガース', 'の', 'キャンデー']
['ホールディングス', 'の', '話題']
['ドーナツ', 'は', '量']


 61%|██████▏   | 851/1386 [01:32<00:28, 19.08it/s]

['中村', 'や', '香川']
['し', 'て', 'カミング']
['オランダ', 'も', '世界']
['沖縄', 'の', 'オランダ']


 62%|██████▏   | 857/1386 [01:32<00:27, 19.31it/s]

['オランダ', 'で', '展示']
['マート', 'の', '方々']
['ストア', 'の', 'スイーツ']
['声優', 'の', 'fukase']


 62%|██████▏   | 860/1386 [01:32<00:25, 20.30it/s]

['ロココ', 'を', '見る']


 62%|██████▏   | 865/1386 [01:32<00:35, 14.68it/s]

['院', 'が', '買う']


 63%|██████▎   | 868/1386 [01:33<00:30, 16.88it/s]

['letao', 'を', '応援']
['年始', 'に', 'ヨーロッパ']


 63%|██████▎   | 875/1386 [01:33<00:38, 13.19it/s]

['グランパス', 'の', 'フレーバー']
['女優', 'の', 'ジーコ']
['インド', 'で', 'ジーコ']
['イーグルス', 'の', '事故']


 63%|██████▎   | 878/1386 [01:33<00:33, 15.36it/s]

['東北', 'が', '桜ん坊']
['ガーデン', 'の', '事故']
['ガーデン', 'の', '鈴木']
['ガーデン', 'の', '鈴木']


 64%|██████▎   | 881/1386 [01:34<00:28, 17.51it/s]

['ガーデン', 'の', '鈴木']


 64%|██████▍   | 887/1386 [01:34<00:27, 18.41it/s]

['孝之', 'の', '東京']
['プロデューサー', 'に', 'テレビ']
['東京', 'の', 'もやもや']


 65%|██████▍   | 896/1386 [01:34<00:22, 22.06it/s]

['アナ', 'と', 'エルサ']
['逢', 'の', '話題']
['pasco', 'の', 'ファン']
['あなた', 'が', 'pasco']
['で', 'は', 'pasco']


 65%|██████▌   | 903/1386 [01:35<00:21, 22.34it/s]

['司会', 'の', '信長']
['。', 'の', '実写']
['スマップ', 'の', 'アナウンサー']
['スマップ', 'の', 'ファースト']


 66%|██████▌   | 910/1386 [01:35<00:18, 26.06it/s]

['神戸', 'と', '裏']
['IKEA', 'は', 'ござい']
['アンパンマンミュージアム', 'は', '神戸']
['IKEA', 'の', '鸚哥']
['IKEA', 'の', '鸚哥']
['アンパンマンミュージアム', 'は', '神戸']
['館', 'の', '鸚哥']


 66%|██████▌   | 916/1386 [01:35<00:18, 25.70it/s]

['神戸', 'の', '京都']
['館', 'の', '鸚哥']
['チャイ', 'は', '商品']
['チャイ', 'は', 'どちら']
['日本', 'と', '愛媛']


 67%|██████▋   | 922/1386 [01:35<00:18, 25.23it/s]

['日本', 'と', '四国']
['ハンバーガー', 'で', 'モ']
['年', 'も', 'MOS']
['リゾート', 'と', '伊勢丹']


 67%|██████▋   | 925/1386 [01:35<00:20, 22.57it/s]

['アナ', 'と', 'エルサ']
['吹き替え', 'や', '歌']
['0', 'の', '恋']
['グループ', 'の', 'akb']
['0', 'の', '軍師']


 67%|██████▋   | 931/1386 [01:36<00:20, 22.63it/s]

['0', 'の', '軍師']
['0', 'の', '国分']
['制作', 'の', '天空']
['選手', 'が', 'ルーム']


 67%|██████▋   | 934/1386 [01:36<00:19, 22.86it/s]

['ストア', 'の', 'ブンデスリーガ']
['今', 'や', '洋食']


 68%|██████▊   | 945/1386 [01:36<00:18, 24.07it/s]

['ランチ', 'も', '食べれる']


 68%|██████▊   | 948/1386 [01:43<04:56,  1.48it/s]

['意外', 'と', '少なく']
['と', 'か', 'スープ']


 69%|██████▉   | 963/1386 [01:44<01:02,  6.81it/s]

['か', 'も', '必須']
['です', 'ね', '一人']
['と', 'か', '天国']


 70%|██████▉   | 969/1386 [01:44<00:41, 10.06it/s]

['なん', 'か', 'サークル']


 71%|███████   | 978/1386 [01:45<00:27, 14.98it/s]

['バンク', 'の', 'ゼロ']


 71%|███████   | 984/1386 [01:45<00:22, 17.85it/s]

['なん', 'か', '響き']


 72%|███████▏  | 992/1386 [01:46<00:22, 17.69it/s]

['しよう', 'かしら', '。']
['深海', 'を', '操作']
['マニピュレーター', 'と', 'か']


 72%|███████▏  | 999/1386 [01:46<00:18, 20.76it/s]

['いう', 'か', '…']


 73%|███████▎  | 1007/1386 [01:46<00:13, 27.43it/s]

['柴犬', 'や', 'と']
['レトリーバー', 'と', '柴']
['レトリーバー', 'と', '柴']
['のばし', 'ちゃ', '駄目']


 74%|███████▎  | 1019/1386 [01:47<00:14, 26.19it/s]

['いう', 'か', '…']


 74%|███████▍  | 1028/1386 [01:47<00:15, 23.16it/s]

['編集', 'と', 'か']
['てる', 'ん', 'でしょう']
['遣い', 'ながら', '吸い']
['です', 'か', '大晦日']


 75%|███████▍  | 1038/1386 [01:48<00:21, 15.99it/s]

['年始', 'も', '0']
['N', 'を', 'y']
['度', 'と', '競演']


 75%|███████▌  | 1044/1386 [01:48<00:19, 17.56it/s]

['夜行', 'も', '面白い']
['グッド', 'だけ', 'です']


 76%|███████▌  | 1053/1386 [01:48<00:15, 21.80it/s]

['因み', 'に', '昨日']


 76%|███████▋  | 1059/1386 [01:49<00:17, 18.37it/s]

['た', 'くらい', 'でしょう']


 77%|███████▋  | 1070/1386 [01:49<00:11, 26.54it/s]

['因み', 'に', 'あなた']
['です', 'かー', '。']
['いつ', 'も', '家']
['暗く', 'まで', 'なり']


 77%|███████▋  | 1074/1386 [01:49<00:10, 29.21it/s]

['一人', 'の', '時']
['意外', 'と', '地味']


 78%|███████▊  | 1081/1386 [01:49<00:12, 23.73it/s]

['と', 'か', '自然']
['定番', 'の', 'クライン']
['ガルボ', 'って', 'しっ']
['プロ', 'に', 'なろう']
['と', 'か', '自然']


 78%|███████▊  | 1088/1386 [01:50<00:13, 22.17it/s]

['大物', 'か', 'も']
['て', 'から', '声掛け']


 79%|███████▊  | 1091/1386 [01:50<00:13, 21.61it/s]

['互い', 'に', '見']
['ん', 'が', '…']
['か', 'も', '必須']
['か', 'が', 'ひどかっ']


 80%|███████▉  | 1102/1386 [01:50<00:11, 24.60it/s]

['酒', 'は', '酎ハイ']
['酒', 'は', '酎ハイ']
['一人', 'の', '方']


 80%|███████▉  | 1108/1386 [01:51<00:11, 24.23it/s]

['指定', 'の', 'かっぱ']
['と', 'か', '表情']
['言葉', 'が', '切磋']
['一人', 'の', '方']
['御陰', 'で', '気づい']


 81%|████████  | 1125/1386 [01:51<00:09, 28.55it/s]

['因み', 'に', '足']
['同士', 'で', '忘年']


 82%|████████▏ | 1133/1386 [01:52<00:08, 28.66it/s]

['なん', 'か', 'サークル']
['手伝い', 'すら', 'ろくに']


 82%|████████▏ | 1141/1386 [01:52<00:07, 32.16it/s]

['ウインズ', 'って', 'あの']
['瀕し', 'て', '手術']


 83%|████████▎ | 1149/1386 [01:52<00:07, 33.79it/s]

['てる', 'ん', 'でしょう']
['な', 'ん', 'でしょう']


 83%|████████▎ | 1157/1386 [01:52<00:07, 32.71it/s]

['け', 'で', '掃除']
['机', 'なんて', '論外']
['と', 'か', '天国']


 84%|████████▍ | 1171/1386 [01:53<00:09, 23.18it/s]

['すみこ', 'の', 'メーク']
['ない', 'ん', 'でしょう']


 85%|████████▌ | 1182/1386 [01:53<00:07, 27.32it/s]

['趣味', 'の', '白土']
['モード', 'の', 'なれ']
['機会', 'と', '気合い']


 87%|████████▋ | 1201/1386 [01:54<00:06, 27.17it/s]

['割', 'も', '夏']


 87%|████████▋ | 1205/1386 [01:54<00:07, 25.26it/s]

['だ', 'か', '取り留め']
['麺類', 'で', '済ま']


 88%|████████▊ | 1214/1386 [01:54<00:05, 32.12it/s]

['です', 'けど', '名店']
['ダンク', 'は', '好き']
['き', 'て', '院生']
['機', 'なんか', 'ダイソン']


 89%|████████▉ | 1235/1386 [01:55<00:06, 23.15it/s]

['対策', 'と', 'か']
['だ', 'と', 'ぼろ負け']


 89%|████████▉ | 1238/1386 [01:56<00:07, 18.57it/s]

['どう', 'か', '分かり']
['しまえ', 'ば', '規則']


 91%|█████████ | 1258/1386 [01:56<00:04, 27.88it/s]

['人類', 'に', '一人']
['いう', 'と', '一人']
['と', 'か', '外']
['飲め', 'て', '食べれ']


 91%|█████████▏| 1266/1386 [01:57<00:04, 29.25it/s]

['対し', 'て', '好き嫌い']


 92%|█████████▏| 1276/1386 [01:57<00:04, 27.36it/s]

['オー', 'と', 'か']


 93%|█████████▎| 1283/1386 [01:57<00:03, 28.42it/s]

['味醂', 'を', '0']
['料理', 'も', 'さっさと']
['に', 'は', '敵い']
['です', 'かー', '。']


 93%|█████████▎| 1287/1386 [01:57<00:03, 28.10it/s]

['です', 'か', '年始']
['ボール', 'は', '食え']
['雑誌', 'が', '0']


 94%|█████████▎| 1298/1386 [01:58<00:03, 28.66it/s]

['か', 'も', '必須']
['と', 'か', '日本']
['部活', 'から', 'ずっと']


 94%|█████████▍| 1304/1386 [01:58<00:02, 27.44it/s]

['練習', 'と', 'か']
['特番', 'で', '長かっ']
['因み', 'に', '何']
['鴨の嘴', 'と', 'いう']


 95%|█████████▍| 1311/1386 [01:58<00:02, 28.68it/s]

['かい', 'の', '時期']
['言葉', 'が', '切磋']


 95%|█████████▌| 1317/1386 [01:58<00:02, 28.06it/s]

['な', 'ん', 'でしょう']
['言葉', 'が', '切磋']
['た', 'から', '…']


 96%|█████████▌| 1327/1386 [01:59<00:02, 28.67it/s]

['時代', 'だけ', 'バス']
['虎', 'と', '黒猫']


 96%|█████████▌| 1333/1386 [01:59<00:02, 24.32it/s]

['ゼミ', 'に', '外国']


 97%|█████████▋| 1340/1386 [01:59<00:01, 27.32it/s]

['相乗り', 'と', 'か']


 97%|█████████▋| 1348/1386 [02:00<00:01, 31.59it/s]

['か', 'は', '作り置き']
['何', 'の', 'アンコール']


 98%|█████████▊| 1356/1386 [02:00<00:01, 24.18it/s]

['さん', 'で', '育た']
['デジカメ', 'は', 'もと']
['た', 'くらい', 'でしょう']


 98%|█████████▊| 1363/1386 [02:00<00:00, 24.24it/s]

['送迎', 'に', '親']
['昼間', 'と', 'か']


 99%|█████████▉| 1371/1386 [02:00<00:00, 29.57it/s]

['因み', 'に', '足']
['因み', 'に', '足']


 99%|█████████▉| 1375/1386 [02:01<00:00, 30.17it/s]

['それ', 'は', '仕方ない']
['作っ', 'て', 'いやいや']


100%|██████████| 1386/1386 [02:01<00:00, 11.40it/s]

['実', 'は', '。']


In [19]:
y_pred_2.count(0)

754

In [20]:
score(y, y_pred_2)

confusion matrix = 
 [[691 292]
 [ 63 340]]
accuracy =  0.7438672438672439
precision =  0.5379746835443038
recall =  0.8436724565756824
f1 score =  0.6570048309178744


- 使い方が難しい

        confusion matrix = 
        [[666 713]
        [  6   1]]
        accuracy =  0.48124098124098125
        precision =  0.0014005602240896359
        recall =  0.14285714285714285
        f1 score =  0.0027739251040221915

- ちょっと改善(プログラムミスを修正)

        confusion matrix = 
        [[870 509]
        [  6   1]]
        accuracy =  0.6284271284271284
        precision =  0.00196078431372549
        recall =  0.14285714285714285
        f1 score =  0.0038684719535783366

    - 意外と普通っぽい ngram がない
        - ["を", "覚える", *]

    - スコア化はせず， ngram が存在しなければアウトと仮定した
        - 先行研究のモデルで，そういったボーダーは存在しない

- 用法エラー + 誤情報

        confusion matrix = 
        [[758 225]
        [118 285]]
        accuracy =  0.7525252525252525
        precision =  0.5588235294117647
        recall =  0.707196029776675
        f1 score =  0.6243154435925521
    

    

In [21]:
data_path = "../X_y_data/base_y_pred/"
# data_name = "utt_form.pickle"
data_name = "utt_content.pickle"
dataM = DataManager(data_path)
dataM.save_data(data_name, [y, y_pred_2])

success save : ../X_y_data/base_y_pred/utt_content.pickle


In [16]:
df = analyzer_unidic.analyze_with_dataframe("頼りがいのあった可愛くない兄さんが死んだ")
df

surface part_of_speech infl_type infl_form base_form  \
0     頼りがい    名詞-普通名詞-一般-                         タヨリガイ   
1        の       助詞-格助詞--                             ノ   
2       あっ     動詞-非自立可能--     五段-ラ行   連用形-促音便        アル   
3        た         助動詞---     助動詞-タ    連体形-一般         タ   
4      可愛く       形容詞-一般--       形容詞    連用形-一般      カワイイ   
5       ない    形容詞-非自立可能--       形容詞    連体形-一般        ナイ   
6        兄    名詞-普通名詞-一般-                            アニ   
7       さん    接尾辞-名詞的-一般-                            サン   
8        が       助詞-格助詞--                             ガ   
9       死ん        動詞-一般--     五段-ナ行   連用形-撥音便        シヌ   
10       だ         助動詞---     助動詞-タ    連体形-一般         タ   

                             reading phonetic  
0   [頼り甲斐, 頼りがい, タヨリガイ, 頼りがい, タヨリガイ]        *  
1                    [の, の, ノ, の, ノ]        *  
2               [有る, あっ, アッ, ある, アル]        *  
3                    [た, た, タ, た, タ]        *  
4        [可愛い, 可愛く, カワイク, 可愛い, カワイー]        *  
5               [無い, ない, ナイ, ない, ナイ]        *  
6                  [兄, 兄, アニ, 兄, アニ]        *  
7               [さん, さん, サン, さん, サン]        *  
8                    [が, が, ガ, が, ガ]        *  
9               [死ぬ, 死ん, シン, 死ぬ, シヌ]        *  
10                   [た, だ, ダ, だ, ダ]        *